pkg/config/feature_gate.go

```diff
@@ -11,6 +11,7 @@ type Feature string
 
 const (
 	EnableListenerRulesTagging Feature = "enable-listener-rules-tagging"
```

Comment:
Lets use `CamelCase` for the feature gate keys instead of `dashed-keys`, for examle
`EnableListenerRulesTagging` instead of `enable-listener-rules-tagging`

This makes it consistent with upstream kubernetes feature gates. It also reduces confusion with the controller command line flags.